darwin/cli_functions.py

```diff
@@ -17,36 +18,49 @@
     ValidationError,
 )
 from darwin.table import Table
-from darwin.utils import find_files, persist_client_configuration, secure_continue_request
+from darwin.utils import find_files, persist_client_configuration, prompt, secure_continue_request
 
 
-def authenticate(email: str, password: str, projects_dir: str) -> Config:
-    """Authenticate user against the server and creates a configuration file for it
+def authenticate(
+    api_key: str,
+    default_team: Optional[bool] = None,
+    datasets_dir: Optional[Path] = None
+) -> Config:
+    """Authenticate the API key against the server and creates a configuration file for it
 
     Parameters
     ----------
-    email : str
-        Email to use for the client login
-    password : str
-        Password to use for the client login
-    projects_dir : str
-         String where the client should be initialized from
+    api_key : str
+        API key to use for the client login
+    default_team: bool
+        Flag to make the team the default one
+    datasets_dir: Path
+        Dataset directory on the file system
 
     Returns
     -------
     Config
     A configuration object to handle YAML files
     """
-    # Resolve the home folder if the project_dir starts with ~ or ~user
-    projects_dir = Path(os.path.expanduser(projects_dir))
-    Path(projects_dir).mkdir(parents=True, exist_ok=True)
-    print(f"Projects directory created {projects_dir}")
+    # Resolve the home folder if the dataset_dir starts with ~ or ~user
 
     try:
-        client = Client.login(email=email, password=password, projects_dir=projects_dir)
+        client = Client.from_api_key(api_key=api_key)
         config_path = Path.home() / ".darwin" / "config.yaml"
         config_path.parent.mkdir(exist_ok=True)
-        return persist_client_configuration(client)
+
+        if default_team is None:
+            default_team = input(f"Make {client.default_team} the default team? [y/N] ") in ["Y", "y"]
+        if datasets_dir is None:
+            datasets_dir = prompt("Datasets directory", "~/.darwin/datasets")
+
+        datasets_dir = Path(os.path.expanduser(datasets_dir))
```

Comment:
We are always trying to use `pathlib` functions in place of `os.path` ones
In this case: https://docs.python.org/3/library/pathlib.html#pathlib.Path.home
```suggestion
        datasets_dir = Path.home() / datasets_dir
```

Reply:
I totally agree with preferring Path, but this is actually not the same thing in this case.
 
`expanduser` replaces `~` with the home path, if there is a `~` in the string, but if there isn't it won't do anything. 

While Path.home() always adds the home directory to the path. (Imagine `datasets_dir = "/datasets/"`)

But I just learned from the docs that Path has `expanduser`


```suggestion
        datasets_dir = Path(datasets_dir).expanduser()
```

Reply:
Ah-ha, I misunderstood what you used `expanduser` for, but still a useful suggestion

![image](https://user-images.githubusercontent.com/8004379/71100658-b18fa600-21b5-11ea-9b73-a403cf14dbd8.png)